대전광역시 날씨와 관광객 분석
--

In [77]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

daejeontour=pd.read_excel('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전.xls')
daejunweather1112=pd.read_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전지방기상청(11-12).csv', parse_dates=['일시'])
daejunweather1314=pd.read_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전지방기상청(13-14).csv', parse_dates=['일시'])
daejunweather1516=pd.read_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전지방기상청(15-16).csv', parse_dates=['일시'])
daejunweather1718=pd.read_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전지방기상청(17-18).csv', parse_dates=['일시'])
daejunweather1920=pd.read_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전지방기상청(19-20).csv', parse_dates=['일시'])

In [78]:
daejeontour.head()

시도  군구       관광지 내/외국인  2011년 01월  2011년 02월  2011년 03월  2011년 04월  \
0  대전광역시  동구  만인산자연휴양림   내국인     4100.0     3900.0     4000.0    24600.0   
1  대전광역시  동구  만인산자연휴양림    합계     4100.0     3900.0     4000.0    24600.0   
2  대전광역시  동구   상소오토캠핑장   내국인        NaN        NaN        NaN        NaN   
3  대전광역시  동구   상소오토캠핑장    합계        NaN        NaN        NaN        NaN   
4  대전광역시  동구     세천유원지   내국인     4100.0     3900.0     4000.0    19800.0   

   2011년 05월  2011년 06월  ...  2019년 03월  2019년 04월  2019년 05월  2019년 06월  \
0    18800.0    13800.0  ...        NaN        NaN        NaN        NaN   
1    18800.0    13800.0  ...        NaN        NaN        NaN        NaN   
2        NaN        NaN  ...     1400.0     1824.0     2620.0     2112.0   
3        NaN        NaN  ...     1400.0     1824.0     2620.0     2112.0   
4    19075.0    13200.0  ...        NaN        NaN        NaN        NaN   

   2019년 07월  2019년 08월  2019년 09월  2019년 10월  2019년 11월  2019년 12월  
0        NaN        NaN        NaN        NaN        NaN        NaN  
1        NaN        NaN        NaN        NaN        NaN        NaN  
2     2348.0     3108.0     1704.0     2840.0        0.0        0.0  
3     2348.0     3108.0     1704.0     2840.0        0.0        0.0  
4        NaN        NaN        NaN        NaN        NaN        NaN  

[5 rows x 112 columns]

In [79]:
daejeonweather=pd.concat([daejunweather1112, daejunweather1314, daejunweather1516, daejunweather1718, daejunweather1920], axis=0)
daejeonweather=daejeonweather.loc[daejeonweather['지점명']=='대전']

In [80]:
daejeonweather.head()

지점 번호 지점명         일시  평균 기온(°C)  최저 기온(°C)  최저 기온 시각(hhmi)  최고 기온(°C)  \
0    133  대전 2011-01-01       -6.8      -12.3           649.0       -0.6   
1    133  대전 2011-01-02       -5.4      -11.0           734.0        2.2   
2    133  대전 2011-01-03       -4.5      -10.3           752.0       -0.1   
3    133  대전 2011-01-04       -2.6       -7.1           639.0        0.6   
4    133  대전 2011-01-05       -0.7       -3.2          2350.0        1.6   

   최고 기온 시각(hhmi)  강수 계속시간(hr)  10분 최다강수량(mm)  ...  0.5m 지중온도(°C)  \
0            1609          NaN            NaN  ...            3.1   
1            1433          NaN            NaN  ...            3.1   
2            1516          NaN            NaN  ...            2.8   
3            1314          NaN            NaN  ...            2.9   
4            1135          NaN            NaN  ...            2.7   

   1.0cm 지중온도(°C)  1.5cm 지중온도(°C)  3.0cm 지중온도(°C)  5.0cm 지중온도(°C)  \
0             6.7             9.5            16.0            18.2   
1             6.9             9.5            16.2            18.5   
2             6.6             9.2            15.9            18.3   
3             6.5             9.2            15.9            18.2   
4             6.2             9.0            15.9            18.1   

   합계 대형증발량(mm)  합계 소형증발량(mm)  9-9강수(mm)  \
0           NaN           1.0        NaN   
1           NaN           0.7        NaN   
2           NaN           0.5        0.0   
3           NaN           0.8        0.0   
4           NaN           1.3        0.0   

                                                일기현상  안개 계속 시간(hr)  
0                                                NaN           NaN  
1  {박무}0715-{박무}{강도1}0900-1010. {박무}2010-{박무}{강도0...           NaN  
2  -{박무}-{박무}{강도0}0300-{박무}{강도0}0600-{박무}{강도0}090...           NaN  
3  -{박무}-{박무}{강도0}0300-{박무}{강도0}0600-0735. {박무}19...           NaN  
4  -{박무}-{박무}{강도0}0300-{박무}{강도0}0600-0620. {눈}010...           NaN  

[5 rows x 62 columns]

In [81]:
daejeonweather.drop(['일기현상'], axis=1, inplace=True)

In [82]:
daejeonweather.columns

Index(['지점 번호', '지점명', '일시', '평균 기온(°C)', '최저 기온(°C)', '최저 기온 시각(hhmi)',
       '최고 기온(°C)', '최고 기온 시각(hhmi)', '강수 계속시간(hr)', '10분 최다강수량(mm)',
       '10분 최다강수량 시각(hhmi)', '1시간 최다강수량(mm)', '1시간 최다강수량 시각(hhmi)', '일강수량(mm)',
       '최대 순간풍속(m/s)', '최대 순간풍속 풍향(16방위)', '최대 순간풍속 시각(hhmi)', '최대 풍속(m/s)',
       '최대 풍속 풍향(16방위)', '최대 풍속 시각(hhmi)', '평균 풍속(m/s)', '풍정합(100m)',
       '최다 풍향(16방위)', '평균 이슬점온도(°C)', '최소 상대습도(%)', '평균 상대습도 시각(hhmi)',
       '평균 상대습도(%)', '평균 중기압(hPa)', '평균 현지기압(hPa)', '최고 해면 기압(hPa)',
       '최고 해면기압 시각(hhmi)', '최저 해면기압(hPa)', '최저 해면기압 시각(hhmi)', '평균 해면기압(hPa)',
       '가조시간(hr)', '합계 일조 시간(hr)', '1시간 최다 일사 시각(hhmi)', '1시간 최다 일사량(MJ/m2)',
       '합계 일사량(MJ/m2)', '일 최심신적설(cm)', '일 최심신적설 시각(hhmi)', '일 최심적설(cm)',
       '일 최심적설 시각(hhmi)', '합계 3시간 신적설(cm)', '평균 전운량(10분위)', '평균 중하층운량(10분위)',
       '평균 지면온도(°C)', '최저 초상온도(°C)', '평균 5cm 지중온도(°C)', '평균 10cm 지중온도(°C)',
       '평균 20cm 지중온도(°C)', '평균 30cm 지중온도(°C)', '0.5m 지중온도(°C)',
       '1.0cm 지중온도(°C)', '1.5cm 지중온도(°C)'

시간칼럼 분류
--

In [83]:
#daejeonweather['time'] = daejeonweather['일시'].dt.strftime('%Y%m%d')
daejeonweather['year_month'] = daejeonweather['일시'].dt.strftime('%Y%m')
#daejeonweather['month_day'] = daejeonweather['일시'].dt.strftime('%m%d')
daejeonweather['year'] = daejeonweather['일시'].dt.strftime('%Y')
#daejeonweather['month']=daejeonweather['일시'].dt.strftime('%m')

#daejeonweather['time']=daejeonweather['time'].astype(int) 
daejeonweather['year_month']=daejeonweather['year_month'].astype(int)
#daejeonweather['month_day']=daejeonweather['month_day'].astype(int) 
daejeonweather['year']=daejeonweather['year'].astype(int) 
#daejeonweather['month']=daejeonweather['month'].astype(int)

In [84]:
daejeonweather.isna().sum()

지점 번호              0
지점명                0
일시                 0
평균 기온(°C)          0
최저 기온(°C)          0
                ... 
합계 소형증발량(mm)       5
9-9강수(mm)       2146
안개 계속 시간(hr)    3520
year_month         0
year               0
Length: 63, dtype: int64

In [85]:
daejeonweather=daejeonweather.fillna(0)

In [86]:
daejeonweather.drop(['지점 번호','지점명'], axis=1, inplace=True)

2011년 1월부터 2019년 12월까지 월별로 데이터를 정리합니다.
--

In [89]:
daejeonweather=daejeonweather.loc[daejeonweather['year']<2020]

In [90]:
daejeonweather=daejeonweather.groupby(['year_month'])['평균 기온(°C)', '최저 기온(°C)', '최저 기온 시각(hhmi)', '최고 기온(°C)',
       '최고 기온 시각(hhmi)', '강수 계속시간(hr)', '10분 최다강수량(mm)', '10분 최다강수량 시각(hhmi)',
       '1시간 최다강수량(mm)', '1시간 최다강수량 시각(hhmi)', '일강수량(mm)', '최대 순간풍속(m/s)',
       '최대 순간풍속 풍향(16방위)', '최대 순간풍속 시각(hhmi)', '최대 풍속(m/s)', '최대 풍속 풍향(16방위)',
       '최대 풍속 시각(hhmi)', '평균 풍속(m/s)', '풍정합(100m)', '최다 풍향(16방위)',
       '평균 이슬점온도(°C)', '최소 상대습도(%)', '평균 상대습도 시각(hhmi)', '평균 상대습도(%)',
       '평균 중기압(hPa)', '평균 현지기압(hPa)', '최고 해면 기압(hPa)', '최고 해면기압 시각(hhmi)',
       '최저 해면기압(hPa)', '최저 해면기압 시각(hhmi)', '평균 해면기압(hPa)', '가조시간(hr)',
       '합계 일조 시간(hr)', '1시간 최다 일사 시각(hhmi)', '1시간 최다 일사량(MJ/m2)',
       '합계 일사량(MJ/m2)', '일 최심신적설(cm)', '일 최심신적설 시각(hhmi)', '일 최심적설(cm)',
       '일 최심적설 시각(hhmi)', '합계 3시간 신적설(cm)', '평균 전운량(10분위)', '평균 중하층운량(10분위)',
       '평균 지면온도(°C)', '최저 초상온도(°C)', '평균 5cm 지중온도(°C)', '평균 10cm 지중온도(°C)',
       '평균 20cm 지중온도(°C)', '평균 30cm 지중온도(°C)', '0.5m 지중온도(°C)',
       '1.0cm 지중온도(°C)', '1.5cm 지중온도(°C)', '3.0cm 지중온도(°C)', '5.0cm 지중온도(°C)',
       '합계 대형증발량(mm)', '합계 소형증발량(mm)', '9-9강수(mm)', '안개 계속 시간(hr)'].mean().reset_index()

In [91]:
daejeonweather.head()

year_month  평균 기온(°C)  최저 기온(°C)  최저 기온 시각(hhmi)  최고 기온(°C)  \
0      201101  -5.706452 -10.370968     1068.967742  -0.590323   
1      201102   1.771429  -3.310714      811.464286   7.642857   
2      201103   4.496774  -0.661290      864.612903  10.493548   
3      201104  11.636667   5.856667      816.800000  17.733333   
4      201105  18.145161  13.141935     1007.612903  23.580645   

   최고 기온 시각(hhmi)  강수 계속시간(hr)  10분 최다강수량(mm)  10분 최다강수량 시각(hhmi)  \
0     1370.387097     0.000000       0.000000            0.000000   
1     1451.071429     0.000000       0.000000            0.000000   
2     1364.935484     0.000000       0.000000            0.000000   
3     1361.000000     0.166667     172.200000            0.723333   
4     1349.322581     0.612903     188.709677            1.806452   

   1시간 최다강수량(mm)  ...  평균 30cm 지중온도(°C)  0.5m 지중온도(°C)  1.0cm 지중온도(°C)  \
0       0.000000  ...          1.083871       1.522581        5.180645   
1       0.000000  ...          1.860714       1.275000        3.664286   
2       0.000000  ...          6.725806       6.316129        6.719355   
3     299.933333  ...         12.650000      11.980000       10.963333   
4     181.645161  ...         18.254839      17.974194       16.141935   

   1.5cm 지중온도(°C)  3.0cm 지중온도(°C)  5.0cm 지중온도(°C)  합계 대형증발량(mm)  합계 소형증발량(mm)  \
0        7.800000       14.638710       17.758065      0.000000      0.987097   
1        5.635714       12.014286       16.610714      0.000000      1.110714   
2        7.161290       10.725806       15.354839      0.000000      2.519355   
3       10.426667       11.110000       14.500000      2.553333      3.430000   
4       14.919355       12.741935       14.193548      3.109677      4.045161   

   9-9강수(mm)  안개 계속 시간(hr)  
0   0.129032         0.000  
1   1.707143         0.265  
2   0.516129         0.000  
3   2.466667         0.000  
4   5.290323         0.000  

[5 rows x 59 columns]

대전시 관광지 관광객도 내국인 한정으로 정리합니다.
--

In [92]:
daejeontour=daejeontour.loc[daejeontour['내/외국인']=='내국인']

In [93]:
daejeontour['관광지'].unique()

array(['만인산자연휴양림', '상소오토캠핑장', '세천유원지', '식장산', '우암사적공원', '찬샘마을',
       '코스모스관광호텔', '태웅관광호텔', '한밭교육박물관', '한일관광호텔', '대전 아쿠아월드', '대전오월드',
       '보문산', '뿌리공원', '남선공원종합체육관', '대전시립미술관', '대전예술의전당', '장태산 자연휴양림',
       '장태산자연휴양림', '한밭수목원', '㈜제이에이치레전드호텔', '계룡산 수통골', '계룡스파텔 대온천탕',
       '국립중앙과학관', '국립현충원', '꿈돌이랜드', '대전시민천문대', '대전어린이회관', '솔로몬 로파크',
       '엑스포과학공원', '유성유스호스텔', '유성코리아온천관광호텔', '유성호텔', '지질박물관', '호텔 인터시티',
       '호텔리베라', '호텔아드리아', '홍인관광호텔', '화폐박물관', '대청호', '로하스가족공원 캠핑장',
       '장동삼림욕장'], dtype=object)

In [94]:
len(daejeontour['관광지'].unique())

42

In [95]:
daejeontour.drop(['시도','군구','내/외국인'], axis=1, inplace=True)

In [96]:
daejeontour=daejeontour.set_index('관광지')

In [97]:
daejeontour=daejeontour.transpose()

In [98]:
daejeontour

관광지        만인산자연휴양림  상소오토캠핑장    세천유원지      식장산   우암사적공원     찬샘마을  코스모스관광호텔  \
2011년 01월    4100.0      NaN   4100.0   3100.0   4100.0      NaN      10.0   
2011년 02월    3900.0      NaN   3900.0   5050.0   3900.0      NaN      10.0   
2011년 03월    4000.0      NaN   4000.0   4000.0   4000.0      NaN      10.0   
2011년 04월   24600.0      NaN  19800.0  15300.0  24810.0      NaN     400.0   
2011년 05월   18800.0      NaN  19075.0  14150.0  17800.0      NaN     480.0   
...             ...      ...      ...      ...      ...      ...       ...   
2019년 08월       NaN   3108.0      NaN      NaN      NaN    818.0       NaN   
2019년 09월       NaN   1704.0      NaN      NaN      NaN   4085.0       NaN   
2019년 10월       NaN   2840.0      NaN      NaN      NaN  10474.0       NaN   
2019년 11월       NaN      0.0      NaN      NaN      NaN    408.0       NaN   
2019년 12월       NaN      0.0      NaN      NaN      NaN     93.0       NaN   

관광지        태웅관광호텔  한밭교육박물관  한일관광호텔  ...     유성호텔   지질박물관  호텔 인터시티    호텔리베라  \
2011년 01월   100.0    100.0    10.0  ...  40473.0  7375.0  44787.0  47905.0   
2011년 02월   100.0    226.0     5.0  ...  38845.0  3635.0  40803.0  40586.0   
2011년 03월   100.0    516.0   490.0  ...  46757.0  2789.0  44508.0  41767.0   
2011년 04월   490.0    211.0   360.0  ...  83440.0  6840.0  43939.0  44417.0   
2011년 05월   480.0    360.0   300.0  ...  44156.0  6690.0  40967.0  40340.0   
...           ...      ...     ...  ...      ...     ...      ...      ...   
2019년 08월     NaN   1647.0     NaN  ...      NaN  6209.0      NaN      NaN   
2019년 09월     NaN   5491.0     NaN  ...      NaN  2911.0      NaN      NaN   
2019년 10월     NaN   2813.0     NaN  ...      NaN  4221.0      NaN      NaN   
2019년 11월     NaN   2258.0     NaN  ...      NaN  4548.0      NaN      NaN   
2019년 12월     NaN   2312.0     NaN  ...      NaN  3241.0      NaN      NaN   

관광지         호텔아드리아   홍인관광호텔    화폐박물관      대청호  로하스가족공원 캠핑장   장동삼림욕장  
2011년 01월  56653.0  33687.0  12515.0  55909.0          NaN  76208.0  
2011년 02월  50201.0  27747.0   7239.0  36450.0          NaN  40650.0  
2011년 03월  45983.0  25328.0   9634.0  16470.0          NaN  23205.0  
2011년 04월  42130.0  33158.0  20600.0  28065.0          NaN  31515.0  
2011년 05월  40677.0  26202.0  19021.0  86890.0          NaN  88250.0  
...            ...      ...      ...      ...          ...      ...  
2019년 08월      NaN      NaN  10216.0      NaN       3800.0      NaN  
2019년 09월      NaN      NaN   5131.0      NaN       2836.0      NaN  
2019년 10월      NaN      NaN  20487.0      NaN       3336.0      NaN  
2019년 11월      NaN      NaN   6016.0      NaN       2368.0      NaN  
2019년 12월      NaN      NaN   5975.0      NaN       1732.0      NaN  

[108 rows x 42 columns]

In [99]:
daejeon_weather=daejeonweather.set_index('year_month')

In [100]:
daejeon_weather

평균 기온(°C)  최저 기온(°C)  최저 기온 시각(hhmi)  최고 기온(°C)  최고 기온 시각(hhmi)  \
year_month                                                                    
201101      -5.706452 -10.370968     1068.967742  -0.590323     1370.387097   
201102       1.771429  -3.310714      811.464286   7.642857     1451.071429   
201103       4.496774  -0.661290      864.612903  10.493548     1364.935484   
201104      11.636667   5.856667      816.800000  17.733333     1361.000000   
201105      18.145161  13.141935     1007.612903  23.580645     1349.322581   
...               ...        ...             ...        ...             ...   
201908      27.203226  23.558065      870.838710  31.635484     1353.838710   
201909      22.486667  18.883333      982.900000  26.773333     1357.766667   
201910      16.200000  11.645161      907.838710  21.532258     1322.774194   
201911       8.650000   3.396667     1030.566667  15.173333     1279.533333   
201912       2.554839  -1.822581     1086.677419   7.893548     1278.967742   

            강수 계속시간(hr)  10분 최다강수량(mm)  10분 최다강수량 시각(hhmi)  1시간 최다강수량(mm)  \
year_month                                                                  
201101         0.000000       0.000000            0.000000       0.000000   
201102         0.000000       0.000000            0.000000       0.000000   
201103         0.000000       0.000000            0.000000       0.000000   
201104         0.166667     172.200000            0.723333     299.933333   
201105         0.612903     188.709677            1.806452     181.645161   
...                 ...            ...                 ...            ...   
201908         0.709677     339.645161            1.658065     267.193548   
201909         0.366667     343.666667            1.766667     408.600000   
201910         0.129032     131.032258            0.732258     125.354839   
201911         0.000000       0.000000            0.000000       0.000000   
201912         0.000000       0.000000            0.000000       0.000000   

            1시간 최다강수량 시각(hhmi)  ...  평균 30cm 지중온도(°C)  0.5m 지중온도(°C)  \
year_month                      ...                                    
201101                1.802581  ...          1.083871       1.522581   
201102                1.642143  ...          1.860714       1.275000   
201103                0.957419  ...          6.725806       6.316129   
201104                2.672667  ...         12.650000      11.980000   
201105                2.670000  ...         18.254839      17.974194   
...                        ...  ...               ...            ...   
201908                1.862258  ...         29.129032      28.112903   
201909                3.977000  ...         24.820000      24.586667   
201910                1.564516  ...         19.667742      20.083871   
201911                1.572333  ...         12.693333      13.706667   
201912                1.940645  ...          5.919355       7.112903   

            1.0cm 지중온도(°C)  1.5cm 지중온도(°C)  3.0cm 지중온도(°C)  5.0cm 지중온도(°C)  \
year_month                                                                   
201101            5.180645        7.800000       14.638710       17.758065   
201102            3.664286        5.635714       12.014286       16.610714   
201103            6.719355        7.161290       10.725806       15.354839   
201104           10.963333       10.426667       11.110000       14.500000   
201105           16.141935       14.919355       12.741935       14.193548   
...                    ...             ...             ...             ...   
201908           26.074194       24.338710       20.029032       16.616129   
201909           24.246667       23.886667       21.353333       17.760000   
201910           21.077419       21.677419       21.387097       18.651613   
201911           16.183333       18.030000       20.036667       18.960000   
201912           10.438710       13.125806       17.574194       18.593548   

            합계 대형증발량(mm)  합계 소

In [101]:
len(daejeon_weather.index)

108

In [102]:
len(daejeontour.index)

108

In [112]:
daejeon_weather=daejeon_weather.reset_index()

In [113]:
daejeontour=daejeontour.reset_index()

대전 날씨와 관광지 데이터 합쳐주기
--

In [116]:
daejeon=pd.concat([daejeon_weather, daejeontour], axis=1)
daejeon

year_month  평균 기온(°C)  최저 기온(°C)  최저 기온 시각(hhmi)  최고 기온(°C)  \
0        201101  -5.706452 -10.370968     1068.967742  -0.590323   
1        201102   1.771429  -3.310714      811.464286   7.642857   
2        201103   4.496774  -0.661290      864.612903  10.493548   
3        201104  11.636667   5.856667      816.800000  17.733333   
4        201105  18.145161  13.141935     1007.612903  23.580645   
..          ...        ...        ...             ...        ...   
103      201908  27.203226  23.558065      870.838710  31.635484   
104      201909  22.486667  18.883333      982.900000  26.773333   
105      201910  16.200000  11.645161      907.838710  21.532258   
106      201911   8.650000   3.396667     1030.566667  15.173333   
107      201912   2.554839  -1.822581     1086.677419   7.893548   

     최고 기온 시각(hhmi)  강수 계속시간(hr)  10분 최다강수량(mm)  10분 최다강수량 시각(hhmi)  \
0       1370.387097     0.000000       0.000000            0.000000   
1       1451.071429     0.000000       0.000000            0.000000   
2       1364.935484     0.000000       0.000000            0.000000   
3       1361.000000     0.166667     172.200000            0.723333   
4       1349.322581     0.612903     188.709677            1.806452   
..              ...          ...            ...                 ...   
103     1353.838710     0.709677     339.645161            1.658065   
104     1357.766667     0.366667     343.666667            1.766667   
105     1322.774194     0.129032     131.032258            0.732258   
106     1279.533333     0.000000       0.000000            0.000000   
107     1278.967742     0.000000       0.000000            0.000000   

     1시간 최다강수량(mm)  ...     유성호텔   지질박물관  호텔 인터시티    호텔리베라   호텔아드리아   홍인관광호텔  \
0         0.000000  ...  40473.0  7375.0  44787.0  47905.0  56653.0  33687.0   
1         0.000000  ...  38845.0  3635.0  40803.0  40586.0  50201.0  27747.0   
2         0.000000  ...  46757.0  2789.0  44508.0  41767.0  45983.0  25328.0   
3       299.933333  ...  83440.0  6840.0  43939.0  44417.0  42130.0  33158.0   
4       181.645161  ...  44156.0  6690.0  40967.0  40340.0  40677.0  26202.0   
..             ...  ...      ...     ...      ...      ...      ...      ...   
103     267.193548  ...      NaN  6209.0      NaN      NaN      NaN      NaN   
104     408.600000  ...      NaN  2911.0      NaN      NaN      NaN      NaN   
105     125.354839  ...      NaN  4221.0      NaN      NaN      NaN      NaN   
106       0.000000  ...      NaN  4548.0      NaN      NaN      NaN      NaN   
107       0.000000  ...      NaN  3241.0      NaN      NaN      NaN      NaN   

       화폐박물관      대청호  로하스가족공원 캠핑장   장동삼림욕장  
0    12515.0  55909.0          NaN  76208.0  
1     7239.0  36450.0          NaN  40650.0  
2     9634.0  16470.0          NaN  23205.0  
3    20600.0  28065.0          NaN  31515.0  
4    19021.0  86890.0          NaN  88250.0  
..       ...      ...          ...      ...  
103  10216.0      NaN       3800.0      NaN  
104   5131.0      NaN       2836.0      NaN  
105  20487.0      NaN       3336.0      NaN  
106   6016.0      NaN       2368.0      NaN  
107   5975.0      NaN       1732.0      NaN  

[108 rows x 102 columns]

In [129]:
daejeon.drop(['index'], axis=1, inplace=True)

In [134]:
len(daejeon.columns)

101

In [136]:
daejeon=daejeon.set_index('year_month')

In [138]:
daejeon.columns[58] # 59번째 칼럼부터 관광지가 나옴

'만인산자연휴양림'

In [148]:
#daejeon.to_csv('C:/Users/bella/Desktop/분석프로젝트/지역별날씨와관광객/대전과세종/대전기후와관광분석.csv', index=False)

만인산자연휴양림
--

In [245]:
daejeon.iloc[:, 67]

year_month
201101     10.0
201102      5.0
201103    490.0
201104    360.0
201105    300.0
          ...  
201908      NaN
201909      NaN
201910      NaN
201911      NaN
201912      NaN
Name: 한일관광호텔, Length: 108, dtype: float64

In [246]:
daejeon[daejeon.iloc[:, 67].notna()]

평균 기온(°C)  최저 기온(°C)  최저 기온 시각(hhmi)  최고 기온(°C)  최고 기온 시각(hhmi)  \
year_month                                                                    
201101      -5.706452 -10.370968     1068.967742  -0.590323     1370.387097   
201102       1.771429  -3.310714      811.464286   7.642857     1451.071429   
201103       4.496774  -0.661290      864.612903  10.493548     1364.935484   
201104      11.636667   5.856667      816.800000  17.733333     1361.000000   
201105      18.145161  13.141935     1007.612903  23.580645     1349.322581   
201106      22.706667  18.196667      769.233333  27.830000     1418.533333   
201107      25.693548  23.203226      643.290323  28.990323     1360.483871   
201108      25.812903  23.106452      942.935484  29.251613     1369.354839   
201109      21.220000  17.233333      867.866667  25.903333     1343.700000   

            강수 계속시간(hr)  10분 최다강수량(mm)  10분 최다강수량 시각(hhmi)  1시간 최다강수량(mm)  \
year_month                                                                  
201101         0.000000       0.000000            0.000000       0.000000   
201102         0.000000       0.000000            0.000000       0.000000   
201103         0.000000       0.000000            0.000000       0.000000   
201104         0.166667     172.200000            0.723333     299.933333   
201105         0.612903     188.709677            1.806452     181.645161   
201106         1.566667     424.033333            4.553333     457.633333   
201107         2.580645     762.000000            6.654839     705.870968   
201108         2.419355     563.064516            6.845161     655.322581   
201109         0.666667     237.533333            1.356667     246.433333   

            1시간 최다강수량 시각(hhmi)  ...      유성호텔    지질박물관  호텔 인터시티    호텔리베라  \
year_month                      ...                                        
201101                1.802581  ...   40473.0   7375.0  44787.0  47905.0   
201102                1.642143  ...   38845.0   3635.0  40803.0  40586.0   
201103                0.957419  ...   46757.0   2789.0  44508.0  41767.0   
201104                2.672667  ...   83440.0   6840.0  43939.0  44417.0   
201105                2.670000  ...   44156.0   6690.0  40967.0  40340.0   
201106                3.289333  ...  299315.0   5036.0  28611.0  38976.0   
201107                4.806774  ...   44930.0   7183.0  30561.0  34734.0   
201108                3.250000  ...   58894.0  15094.0  15714.0  35576.0   
201109                1.940000  ...   45140.0  43980.0  12905.0  35371.0   

             호텔아드리아   홍인관광호텔    화폐박물관       대청호  로하스가족공원 캠핑장    장동삼림욕장  
year_month                                                              
201101      56653.0  33687.0  12515.0   55909.0          NaN   76208.0  
201102      50201.0  27747.0   7239.0   36450.0          NaN   40650.0  
201103      45983.0  25328.0   9634.0   16470.0          NaN   23205.0  
201104      42130.0  33158.0  20600.0   28065.0          NaN   31515.0  
201105      40677.0  26202.0  19021.0   86890.0          NaN   88250.0  
201106      13339.0   2384.0  17607.0   95571.0          NaN   92348.0  
201107      37873.0      NaN  23243.0  102590.0          NaN  108516.0  
201108      32288.0      NaN  27700.0  105815.0          NaN  111074.0  
201109      38069.0      NaN  25795.0   96195.0          NaN  116165.0  

[9 rows x 100 columns]

- 1번째 관광지 부터 10개 관광지 결측치 없는 구간 체크하기

1) 만인산자연휴양림: 201101-201109

2) 상소오토캠핑장: 201601-201912

3) 세천유원지: 201101-201109

4) 식장산: 201101-201109

5) 우암사적공원: 201101-201109

6) 찬샘마을: 201701~201912

7) 코스모스관광호텔: 201101-201109

8) 태웅관광호텔:201101-201109

9) 한밭교육박물관: 201101-201912

10) 한일관광호텔: 201101-201109

대부분 2011년 1월부터 2011년 9월까지만 데이터가 있는 것을 확인 할 수 있음.


2011년부터 2019년까지 년별로 분석하기
--

2011년 1월-9월, 2011년 1월- 12월
--

In [263]:
daejun_1109=daejeon.iloc[:9, 58:]
daejun_1109=daejun_1109.dropna(axis=1)
daejun_1109

만인산자연휴양림    세천유원지      식장산   우암사적공원  코스모스관광호텔  태웅관광호텔  한밭교육박물관  \
year_month                                                                   
201101        4100.0   4100.0   3100.0   4100.0      10.0   100.0    100.0   
201102        3900.0   3900.0   5050.0   3900.0      10.0   100.0    226.0   
201103        4000.0   4000.0   4000.0   4000.0      10.0   100.0    516.0   
201104       24600.0  19800.0  15300.0  24810.0     400.0   490.0    211.0   
201105       18800.0  19075.0  14150.0  17800.0     480.0   480.0    360.0   
201106       13800.0  13200.0  17700.0   8325.0     450.0   380.0    320.0   
201107       21650.0  26200.0  10800.0   9700.0     441.0   860.0    354.0   
201108       21300.0  31100.0  13300.0   7775.0     250.0   250.0    724.0   
201109       16115.0  19045.0  16790.0  13500.0     289.0   200.0    520.0   

            한일관광호텔  대전 아쿠아월드     대전오월드  ...  대전시민천문대   엑스포과학공원      유성호텔  \
year_month                              ...                                
201101        10.0    5148.0    7335.0  ...   7012.0   34604.0   40473.0   
201102         5.0   72774.0   26175.0  ...   7268.0   25901.0   38845.0   
201103       490.0   47724.0   65177.0  ...   5527.0   19593.0   46757.0   
201104       360.0   55424.0  171794.0  ...   6110.0  210135.0   83440.0   
201105       300.0   68231.0  213740.0  ...   6812.0  183645.0   44156.0   
201106       250.0   39375.0  103873.0  ...   6723.0   28602.0  299315.0   
201107       205.0   46284.0   50573.0  ...  18118.0   45479.0   44930.0   
201108       310.0   69054.0  118802.0  ...  13178.0   93342.0   58894.0   
201109       160.0   31414.0   98103.0  ...   6550.0   68945.0   45140.0   

              지질박물관  호텔 인터시티    호텔리베라   호텔아드리아    화폐박물관       대청호    장동삼림욕장  
year_month                                                                   
201101       7375.0  44787.0  47905.0  56653.0  12515.0   55909.0   76208.0  
201102       3635.0  40803.0  40586.0  50201.0   7239.0   36450.0   40650.0  
201103       2789.0  44508.0  41767.0  45983.0   9634.0   16470.0   23205.0  
201104       6840.0  43939.0  44417.0  42130.0  20600.0   28065.0   31515.0  
201105       6690.0  40967.0  40340.0  40677.0  19021.0   86890.0   88250.0  
201106       5036.0  28611.0  38976.0  13339.0  17607.0   95571.0   92348.0  
201107       7183.0  30561.0  34734.0  37873.0  23243.0  102590.0  108516.0  
201108      15094.0  15714.0  35576.0  32288.0  27700.0  105815.0  111074.0  
201109      43980.0  12905.0  35371.0  38069.0  25795.0   96195.0  116165.0  

[9 rows x 29 columns]

- 2011년 1월부터 9월까지 했을 때, 생존 칼럼 29개

In [264]:
daejun_1112=daejeon.iloc[:12, 58:]
daejun_1112=daejun_1112.dropna(axis=1)
daejun_1112

한밭교육박물관  대전 아쿠아월드     대전오월드       보문산      뿌리공원  장태산 자연휴양림  \
year_month                                                               
201101        100.0    5148.0    7335.0  104793.0   20995.0    17063.0   
201102        226.0   72774.0   26175.0  117603.0   45813.0    18885.0   
201103        516.0   47724.0   65177.0  204753.0   61862.0    22391.0   
201104        211.0   55424.0  171794.0  316316.0  106730.0    28964.0   
201105        360.0   68231.0  213740.0  331597.0  238818.0    38474.0   
201106        320.0   39375.0  103873.0  311739.0  103788.0    41790.0   
201107        354.0   46284.0   50573.0  321118.0   66655.0    30750.0   
201108        724.0   69054.0  118802.0  333048.0   79091.0    32328.0   
201109        520.0   31414.0   98103.0  440110.0  137105.0    28720.0   
201110       1748.0   47376.0  205701.0  365000.0  278554.0    34772.0   
201111       2325.0   34545.0   60491.0  343200.0   73027.0    30270.0   
201112       1087.0   51324.0   23316.0  137509.0   50383.0    17300.0   

               한밭수목원  ㈜제이에이치레전드호텔  계룡산 수통골   국립중앙과학관  ...    꿈돌이랜드  대전시민천문대  \
year_month                                            ...                     
201101        8556.0       3910.0  25330.0   79251.0  ...  63931.0   7012.0   
201102       19008.0       2930.0  34116.0   73591.0  ...  43725.0   7268.0   
201103       62154.0       3190.0  38547.0   57815.0  ...  24621.0   5527.0   
201104      123567.0       4350.0  36226.0  186133.0  ...  62386.0   6110.0   
201105      188664.0       3815.0  38242.0  163611.0  ...  72477.0   6812.0   
201106      113646.0       3150.0  31435.0   64248.0  ...  21084.0   6723.0   
201107       92435.0       3980.0  36967.0   85487.0  ...  46201.0  18118.0   
201108       90675.0       4820.0  56231.0  132521.0  ...  54683.0  13178.0   
201109      123550.0        800.0  68199.0   67120.0  ...  29880.0   6550.0   
201110      153753.0       7692.0  83798.0  152144.0  ...  53649.0   8111.0   
201111      121023.0       6838.0  56823.0   77321.0  ...  19587.0   7570.0   
201112       23863.0       7049.0  45197.0   70028.0  ...  41034.0   8624.0   

             엑스포과학공원      유성호텔    지질박물관  호텔 인터시티    호텔리베라   호텔아드리아       대청호  \
year_month                                                                     
201101       34604.0   40473.0   7375.0  44787.0  47905.0  56653.0   55909.0   
201102       25901.0   38845.0   3635.0  40803.0  40586.0  50201.0   36450.0   
201103       19593.0   46757.0   2789.0  44508.0  41767.0  45983.0   16470.0   
201104      210135.0   83440.0   6840.0  43939.0  44417.0  42130.0   28065.0   
201105      183645.0   44156.0   6690.0  40967.0  40340.0  40677.0   86890.0   
201106       28602.0  299315.0   5036.0  28611.0  38976.0  13339.0   95571.0   
201107       45479.0   44930.0   7183.0  30561.0  34734.0  37873.0  102590.0   
201108       93342.0   58894.0  15094.0  15714.0  35576.0  32288.0  105815.0   
201109       68945.0   45140.0  43980.0  12905.0  35371.0  38069.0   96195.0   
201110      242946.0   57287.0   8438.0  39076.0  44472.0  65547.0  108529.0   
201111       35635.0   54618.0   6514.0  41905.0  42944.0  50584.0   78063.0   
201112       28613.0   48235.0   3389.0  42064.0  55244.0  70399.0   57715.0   

              장동삼림욕장  
year_month            
201101       76208.0  
201102       40650.0  
201103       23205.0  
201104       31515.0  
201105       88250.0  
201106       92348.0  
201107      108516.0  
201108      111074.0  
201109      116165.0  
201110      132242.0  
201111      116597.0  
201112       56236.0  

[12 rows x 21 columns]

- 2011년 1월부터 12월까지 했을 때, 생존 칼럼 21개

2012년
--

In [265]:
daejun_12=daejeon.iloc[12:24, 58:]
daejun_12=daejun_12.dropna(axis=1)
daejun_12

한밭교육박물관     대전오월드       보문산      뿌리공원  장태산 자연휴양림     한밭수목원  \
year_month                                                               
201201       1022.0   22855.0  106588.0   42388.0    21690.0   45420.0   
201202        922.0   30236.0  137302.0   47701.0    20358.0   48520.0   
201203       1372.0   58274.0  198652.0   81966.0    18380.0   76454.0   
201204       1798.0  178078.0  336570.0  149334.0    28850.0  158835.0   
201205       1571.0  233268.0  332495.0  354080.0    40979.0  181912.0   
201206        950.0  101597.0  310355.0  132305.0    38740.0   82985.0   
201207       1662.0   73703.0  386452.0  111124.0    30160.0   51788.0   
201208       2772.0  118977.0  399732.0   89100.0    27193.0   45211.0   
201209       5144.0   96672.0  502390.0  106110.0    24960.0   90410.0   
201210       3351.0  215957.0  379000.0  150877.0    32728.0  134015.0   
201211       4578.0   56891.0  345207.0   75389.0    17964.0   53373.0   
201212       2495.0   22903.0  129514.0   51533.0     6938.0   22585.0   

            ㈜제이에이치레전드호텔   계룡산 수통골   국립중앙과학관     국립현충원  ...   엑스포과학공원  유성유스호스텔  \
year_month                                             ...                      
201201           3340.0   50016.0   70672.0  195930.0  ...   37470.0    795.0   
201202           4850.0   52422.0   79457.0  126699.0  ...   35211.0   3362.0   
201203           5650.0   66299.0   66618.0  185294.0  ...   34109.0   1496.0   
201204           5400.0   73153.0  140699.0  212552.0  ...   36531.0   2263.0   
201205           4320.0   78622.0  130424.0  278100.0  ...  346505.0   2319.0   
201206           4650.0   65495.0   85177.0  430900.0  ...   26531.0   2400.0   
201207           3470.0   40562.0   96091.0  157600.0  ...   30981.0   2512.0   
201208           3590.0   30892.0  156430.0  139400.0  ...   40031.0   2912.0   
201209           1800.0   49284.0   94970.0  271000.0  ...   25929.0   1596.0   
201210           2100.0  103101.0  156689.0  268250.0  ...  396931.0   2686.0   
201211           2300.0   62412.0  124527.0  186500.0  ...  147806.0   1814.0   
201212           2050.0   51968.0   94209.0  141750.0  ...   72933.0   1983.0   

               유성호텔    지질박물관  호텔 인터시티    호텔리베라   호텔아드리아    화폐박물관      대청호  \
year_month                                                                  
201201      40268.0   6310.0  39591.0  37974.0  66360.0  20620.0  54987.0   
201202      47890.0   4454.0  42939.0  43631.0  57865.0  21226.0  48883.0   
201203      42255.0   3575.0  40207.0  42175.0  56384.0  23202.0  61413.0   
201204      49241.0   5031.0  39477.0  39233.0  73062.0  32940.0  64886.0   
201205      57975.0   6858.0  34850.0  35947.0  33561.0  31333.0  70034.0   
201206      40597.0   5110.0  35073.0  34516.0  28613.0  21940.0  75000.0   
201207      44740.0   6830.0  36679.0  40836.0  36943.0  18600.0  75172.0   
201208      48756.0  11640.0  28821.0  30956.0  94948.0  22405.0  77378.0   
201209      39222.0   5600.0  31072.0  40166.0  26665.0  20600.0  83550.0   
201210      51649.0  12375.0  35454.0  42304.0  38019.0  30300.0  99816.0   
201211      53644.0   7550.0  41240.0  45751.0  43372.0  27900.0  87480.0   
201212      44532.0   3275.0  37383.0  51469.0  48568.0  23900.0  55363.0   

              장동삼림욕장  
year_month            
201201       53105.0  
201202       45984.0  
201203       59439.0  
201204       67643.0  
201205       74109.0  
201206      105000.0  
201207       86821.0  
201208       91153.0  
201209       99975.0  
201210      132414.0  
201211      118210.0  
201212       59038.0  

[12 rows x 21 columns]

- 2012년에 결측치 없는 데이터 장소 21개

2013년
--

In [266]:
daejun_13=daejeon.iloc[24:36, 58:]
daejun_13=daejun_13.dropna(axis=1)
daejun_13

한밭교육박물관     대전오월드  계룡산 수통골   국립중앙과학관   엑스포과학공원  유성유스호스텔   지질박물관  \
year_month                                                                    
201301       3718.0   27021.0  33316.0   90987.0   29027.0   2356.0  6125.0   
201302       1466.0   33050.0  34368.0   68603.0   17734.0   1625.0  4315.0   
201303       1812.0  123103.0  71887.0   77790.0   36437.0   1645.0  4745.0   
201304       3486.0  144151.0  45631.0  140613.0   36335.0   2310.0  6035.0   
201305       2335.0  219020.0  67434.0  130455.0  209774.0   2934.0  6760.0   
201306       2663.0  105733.0  67050.0   87357.0   26533.0   1947.0  5028.0   
201307       2470.0   66757.0  30912.0   86107.0   43260.0  15470.0  7314.0   
201308       4140.0   90270.0  53199.0  118429.0  185699.0  18089.0  9418.0   
201309       7119.0  103979.0  50064.0   71610.0   36860.0   3317.0  4653.0   
201310       2743.0  190428.0  69903.0  132776.0  317596.0   2326.0  8118.0   
201311       3766.0   62541.0  71683.0   80870.0   41639.0   1682.0  6218.0   
201312       1467.0   36487.0  40770.0   65534.0   36552.0   1046.0  4628.0   

              화폐박물관  
year_month           
201301      16040.0  
201302      23940.0  
201303      41700.0  
201304      24580.0  
201305      27740.0  
201306      18745.0  
201307      17009.0  
201308      18351.0  
201309      11328.0  
201310      14763.0  
201311      15749.0  
201312      13788.0

- 2013년에 데이터가 있는 곳은 총 8곳

2014년
--

In [267]:
daejun_14=daejeon.iloc[36:48, 58:]
daejun_14=daejun_14.dropna(axis=1)
daejun_14

한밭교육박물관     대전오월드      뿌리공원  대전시립미술관  계룡산 수통골   국립중앙과학관  솔로몬 로파크  \
year_month                                                                     
201401       2827.0   51599.0   26867.0  12870.0  44285.0   78720.0  17983.0   
201402       2103.0   49712.0   26173.0   6763.0  38649.0   62780.0  14628.0   
201403       1510.0   80084.0   42723.0   9447.0  59423.0   63269.0  20190.0   
201404       1521.0  132133.0   68967.0   2856.0  52941.0  124417.0  19862.0   
201405       1750.0  190247.0   79361.0  11279.0  65298.0  104455.0  30270.0   
201406       1556.0  103606.0   71874.0   7998.0  63598.0   81887.0  26813.0   
201407       2097.0   86297.0   57787.0  54905.0  61064.0   85242.0  25324.0   
201408       3395.0  152613.0   67264.0  69304.0  62320.0  130219.0  39728.0   
201409       5418.0  104902.0  303053.0  32696.0  65338.0   77399.0  24093.0   
201410       3000.0  191570.0   89251.0   3709.0  69224.0  138872.0  16564.0   
201411       2893.0   56983.0   45537.0   2903.0  46891.0   83566.0  12305.0   
201412       3544.0   28529.0   17860.0   5970.0  25103.0   51961.0  13616.0   

             엑스포과학공원  유성유스호스텔    지질박물관    화폐박물관  
year_month                                       
201401       37218.0   1578.0   7184.0  13888.0  
201402       23765.0   1461.0   5077.0   7971.0  
201403       32074.0   1277.0   4152.0   9126.0  
201404       25704.0      2.0   4329.0  13403.0  
201405      137636.0   1408.0   5979.0  10244.0  
201406        9367.0   1140.0   5064.0   9212.0  
201407        9314.0   1983.0   6982.0   9806.0  
201408       14934.0   1967.0  11905.0  17290.0  
201409        7968.0    923.0   3921.0   6878.0  
201410       12827.0   2307.0   8300.0  14079.0  
201411       13470.0   1463.0   8916.0   9992.0  
201412        1968.0   1360.0   5064.0   7305.0

- 2014년에 데이터가 있는 곳은 총 11곳, 뿌리공원, 대전시립미술관, 솔로몬 로파크가 새로 생김

2015년
--

In [268]:
daejun_15=daejeon.iloc[48:60, 58:]
daejun_15=daejun_15.dropna(axis=1)
daejun_15

한밭교육박물관     대전오월드      뿌리공원  대전시립미술관  계룡산 수통골   국립중앙과학관  솔로몬 로파크  \
year_month                                                                     
201501       2445.0   58794.0   23119.0  12028.0  38484.0   89103.0  21809.0   
201502       1353.0   46509.0   28806.0   4365.0  35656.0   57199.0  18430.0   
201503       1816.0   90662.0   48172.0   6745.0  49599.0   54009.0  21899.0   
201504       1923.0  154251.0   76292.0   6253.0  45135.0  134854.0  31719.0   
201505       2202.0  214227.0  124745.0   8016.0  60337.0  172340.0  27405.0   
201506       1274.0   19441.0   54028.0   9328.0  55443.0   28420.0   5024.0   
201507       1753.0   69977.0   53961.0  15810.0  68884.0   82464.0  11419.0   
201508       1986.0  143018.0   70207.0  27325.0  77394.0  144297.0  24338.0   
201509       6763.0   95595.0   77370.0  19300.0  79578.0  111137.0  13055.0   
201510       2593.0  164997.0  427492.0  28026.0  62875.0  169335.0  22152.0   
201511       2631.0   42453.0   40846.0  25239.0  39625.0   82719.0  30656.0   
201512       3219.0   50811.0   28133.0  20620.0  35362.0   88599.0  32061.0   

            유성유스호스텔    지질박물관    화폐박물관  
year_month                             
201501       1668.0  10652.0  12802.0  
201502       1015.0   5357.0   7642.0  
201503       1496.0   4708.0   9747.0  
201504       2634.0   7775.0  33325.0  
201505       2377.0   9237.0  18619.0  
201506        520.0   1801.0   1916.0  
201507       1556.0   6356.0   6606.0  
201508       2071.0  11129.0  15993.0  
201509       2290.0   5509.0   7612.0  
201510       2357.0  10263.0  14449.0  
201511       2041.0   8792.0  10316.0  
201512       1317.0   6602.0   7740.0

- 엑스포과학공원 사라짐

2016년
--

In [269]:
daejun_16=daejeon.iloc[60:72, 58:]
daejun_16=daejun_16.dropna(axis=1)
daejun_16

상소오토캠핑장  한밭교육박물관     대전오월드      뿌리공원  대전시립미술관  계룡산 수통골   국립중앙과학관  \
year_month                                                                     
201601       1152.0   2658.0   57412.0   27118.0   6172.0  34757.0  115739.0   
201602       1224.0   1406.0   44760.0   26667.0   6869.0  29244.0   75597.0   
201603       1036.0   1847.0   68696.0   56121.0  10602.0  42916.0   62435.0   
201604       1928.0   1949.0  164718.0   89017.0  14531.0  47233.0  158845.0   
201605       2520.0   2067.0  178514.0  112742.0  18375.0  58645.0  152690.0   
201606       1882.0   2213.0   78803.0   92246.0  23398.0  42447.0   82793.0   
201607       2628.0   2213.0   56733.0   58627.0   7410.0  49970.0   94269.0   
201608       3420.0   2302.0   99989.0   58303.0  15322.0  49234.0  118193.0   
201609       2300.0   5980.0   89962.0  398298.0   9434.0  46096.0  122156.0   
201610       2656.0   2420.0  170402.0   83470.0   9546.0  55467.0  163171.0   
201611       1795.0   2395.0   64108.0   49835.0   8677.0  49035.0   78827.0   
201612       1773.0   3232.0   41299.0   28477.0   6886.0  51925.0   83824.0   

            솔로몬 로파크  유성유스호스텔    지질박물관    화폐박물관  
year_month                                      
201601      19463.0   1359.0  10779.0  12794.0  
201602      14623.0    934.0   6079.0   6410.0  
201603      20074.0   1060.0   6591.0   6849.0  
201604      33002.0   1381.0   9034.0  25296.0  
201605      26392.0   1274.0   7644.0  12766.0  
201606      18286.0   1233.0   6753.0   6515.0  
201607      16375.0   1531.0   8907.0  10148.0  
201608      22313.0   2066.0  12721.0  14461.0  
201609      17322.0    259.0   6914.0   8544.0  
201610      26403.0   2034.0   8539.0  11638.0  
201611      20902.0   1880.0   7660.0   7032.0  
201612      28161.0   1682.0   6469.0   7076.0

- 상소오토캠핑장이 생김

2017년
--

In [270]:
daejun_17=daejeon.iloc[72:84, 58:]
daejun_17=daejun_17.dropna(axis=1)
daejun_17

상소오토캠핑장    찬샘마을  한밭교육박물관  대전 아쿠아월드     대전오월드      뿌리공원  남선공원종합체육관  \
year_month                                                                      
201701       1082.0   216.0   2127.0   21392.0   45628.0   36457.0    39998.0   
201702        747.0   140.0   1821.0   22697.0   56167.0   43296.0    34627.0   
201703       1164.0   112.0   1729.0   18372.0   77761.0   64603.0    30768.0   
201704       1564.0  3672.0   2125.0   13350.0  193392.0   97662.0    30461.0   
201705       2707.0  1203.0   2232.0   17607.0  168448.0  100959.0    31637.0   
201706       2116.0  1778.0   2179.0   13876.0   68545.0   65352.0    33628.0   
201707       1850.0   563.0   2031.0   20053.0   42755.0   48990.0    43283.0   
201708       3078.0   406.0   2814.0   26185.0   90553.0   60511.0    47640.0   
201709       2353.0  4551.0   6957.0   11600.0  110542.0  431223.0    33908.0   
201710       3080.0  7673.0   2732.0   14854.0  173895.0  119307.0    23392.0   
201711        972.0  2117.0   2853.0    9340.0   40017.0   56487.0    27304.0   
201712       1500.0   810.0   2040.0   17557.0   28776.0   29226.0    46706.0   

            대전시립미술관  장태산자연휴양림   계룡산 수통골   국립중앙과학관  대전시민천문대  솔로몬 로파크  유성유스호스텔  \
year_month                                                                     
201701       6968.0   42206.0   64544.0   87351.0   6188.0  11581.0   3417.0   
201702       7018.0   39224.0   59022.0   93735.0   9807.0  30948.0   3602.0   
201703       8780.0   57887.0   84891.0   86977.0   6642.0  49246.0   2766.0   
201704       9363.0   79833.0   92041.0  169484.0   7699.0  37547.0   2601.0   
201705      20834.0  107288.0   90042.0  200587.0  10151.0  24215.0   3904.0   
201706       6429.0  103092.0   87367.0  115329.0   6202.0  20928.0   3042.0   
201707      13773.0   79033.0   80989.0  118794.0   8573.0  21575.0   1245.0   
201708      20927.0   52611.0   89928.0  168022.0  25076.0  24007.0   1777.0   
201709      11202.0   67570.0   90054.0  167538.0  10867.0  26150.0   1443.0   
201710       9163.0  103917.0  106922.0  155781.0  14476.0  19199.0   1289.0   
201711       8690.0   93830.0   74575.0   83527.0   6879.0  19959.0   1736.0   
201712      11230.0   34190.0   51146.0   82816.0   6817.0  17257.0    778.0   

             지질박물관    화폐박물관  로하스가족공원 캠핑장  
year_month                                
201701      6249.0   9613.0       1500.0  
201702      7606.0   8902.0       1500.0  
201703      4669.0   7001.0       2292.0  
201704      9186.0  31238.0       2768.0  
201705      6489.0  11244.0       3828.0  
201706      6294.0   7074.0       3532.0  
201707      9108.0   7535.0       2800.0  
201708      9877.0  13598.0       4020.0  
201709      6231.0   8208.0       3676.0  
201710      6752.0   9158.0       3568.0  
201711      7922.0   7158.0       1720.0  
201712      4376.0   6849.0       1656.0

- 찬샘마을, 대전아쿠아월드, 남산공원종합체육관, 장태산자연휴양림, 대전시민천문대, 로하스가족공원 캠핑장 생김

2018년
--

In [272]:
daejun_18=daejeon.iloc[84:96, 58:]
daejun_18=daejun_18.dropna(axis=1)
daejun_18

상소오토캠핑장    찬샘마을  한밭교육박물관  대전 아쿠아월드     대전오월드      뿌리공원  남선공원종합체육관  \
year_month                                                                      
201801       1158.0   194.0   1841.0   21416.0   36102.0   28466.0    34781.0   
201802       1270.0   127.0    944.0   16465.0   37297.0   29207.0    32276.0   
201803       1949.0    58.0   1703.0   20173.0   70098.0   63908.0    27115.0   
201804       2176.0  2898.0   2126.0   14150.0  131949.0   85017.0    24708.0   
201805       2720.0  1799.0   2215.0   19182.0  166951.0  109965.0    27163.0   
201806       3293.0  1587.0   1937.0   14431.0   81258.0  102222.0    29048.0   
201807       2300.0   317.0   1819.0   17430.0   43130.0   68122.0    41256.0   
201808       2783.0   239.0   2256.0   24006.0   95899.0   64612.0    42809.0   
201809       2492.0  3423.0   6132.0   11231.0  105386.0  120148.0    21444.0   
201810       1880.0  9336.0   3076.0   12003.0  143449.0  305951.0    22916.0   
201811       1239.0   912.0   2524.0    9702.0   46678.0   55556.0    23518.0   
201812       1395.0    98.0   2221.0   15046.0   54869.0   35718.0    30855.0   

            대전시립미술관  장태산자연휴양림   계룡산 수통골   국립중앙과학관  대전시민천문대  대전어린이회관  솔로몬 로파크  \
year_month                                                                     
201801       7318.0   53525.0   48454.0   93508.0  12035.0  37838.0  13692.0   
201802       6329.0   31501.0   47935.0   82886.0   5410.0  28966.0  11180.0   
201803       6754.0   44887.0   82958.0   86110.0   8949.0  26358.0  19321.0   
201804      10438.0   42150.0   82580.0  142660.0   6692.0  41807.0  32518.0   
201805      13405.0   67009.0  103115.0  173993.0   9080.0  40514.0   5701.0   
201806      16083.0   54705.0  102884.0  105025.0   7454.0  36425.0  40869.0   
201807       4961.0   42822.0   90036.0  134841.0  13688.0  34938.0  15559.0   
201808      16865.0  101911.0   56749.0  199819.0  22731.0  44103.0  20716.0   
201809      11583.0  129867.0   92125.0  165897.0   8314.0  27895.0  22454.0   
201810       9206.0  123136.0  104157.0  142216.0  15785.0  20778.0  25611.0   
201811       8357.0  103343.0   82765.0   96960.0   7436.0   5344.0  15223.0   
201812      10929.0   44894.0   57394.0   95065.0   6387.0   5614.0  22633.0   

            유성유스호스텔   지질박물관    화폐박물관  로하스가족공원 캠핑장  
year_month                                         
201801       1371.0  5951.0   9776.0       1090.0  
201802        929.0  3389.0   4888.0       1470.0  
201803        816.0  4136.0   6055.0       2020.0  
201804       1348.0  7456.0  21378.0       2688.0  
201805       1342.0  6741.0  11277.0       3428.0  
201806        837.0  4872.0   6394.0       3544.0  
201807       1774.0  5233.0   7326.0       2764.0  
201808       1849.0  8490.0  13375.0       3044.0  
201809       1132.0  4142.0   7291.0       3236.0  
201810       1290.0  5165.0   9172.0       2984.0  
201811        494.0  4602.0   6982.0       1708.0  
201812         56.0  3643.0   7545.0       1864.0

- 대전 어린이회관 생김

2019년
--

In [273]:
daejun_19=daejeon.iloc[96:, 58:]
daejun_19=daejun_19.dropna(axis=1)
daejun_19

상소오토캠핑장     찬샘마을  한밭교육박물관  대전 아쿠아월드     대전오월드      뿌리공원  \
year_month                                                            
201901       1168.0     92.0   1949.0   17035.0   42599.0   39262.0   
201902       1276.0    124.0    808.0   15734.0   46868.0   48279.0   
201903       1400.0    179.0   1368.0   16030.0   51592.0   97916.0   
201904       1824.0   2194.0   1600.0   11197.0  118065.0   88982.0   
201905       2620.0   1878.0   2865.0   20199.0  137642.0  112314.0   
201906       2112.0   1228.0   2170.0   15457.0   76412.0   96430.0   
201907       2348.0    352.0   1878.0   14263.0   34436.0   75645.0   
201908       3108.0    818.0   1647.0   18576.0   57721.0   80405.0   
201909       1704.0   4085.0   5491.0   10143.0   66082.0  420532.0   
201910       2840.0  10474.0   2813.0   12223.0  148600.0   93974.0   
201911          0.0    408.0   2258.0   16075.0   52131.0   61160.0   
201912          0.0     93.0   2312.0   19461.0   26374.0   42741.0   

            남선공원종합체육관  대전시립미술관  장태산자연휴양림   계룡산 수통골   국립중앙과학관  대전시민천문대  \
year_month                                                              
201901        34550.0   6645.0   41007.0   64221.0  104886.0  11892.0   
201902        24025.0   7095.0   45381.0   64373.0   82613.0   8235.0   
201903        25230.0   8495.0   53026.0   49951.0   94794.0   8412.0   
201904        24033.0   7632.0   71516.0   44142.0  127846.0   6413.0   
201905        27042.0   9875.0  119381.0   91269.0  133025.0  16352.0   
201906        27583.0  21493.0  128514.0   96470.0  119210.0   6552.0   
201907        37837.0  10153.0   85810.0   88393.0  111462.0   8410.0   
201908        40400.0  19240.0  107483.0   85577.0  158881.0  14936.0   
201909        19688.0  12330.0   62482.0   78944.0  157633.0   7142.0   
201910        25384.0  10467.0   97764.0  119600.0  137386.0  10651.0   
201911        25481.0   9522.0   99805.0  107300.0   75532.0   7708.0   
201912        33939.0  11588.0   47567.0   72600.0   73972.0   6460.0   

            대전어린이회관  솔로몬 로파크  유성유스호스텔   지질박물관    화폐박물관  로하스가족공원 캠핑장  
year_month                                                           
201901       6279.0  14580.0   1488.0  4817.0   9574.0       1092.0  
201902      25414.0  13595.0    650.0  3541.0   5217.0       1600.0  
201903      46362.0  19957.0    943.0  3925.0   6132.0       1792.0  
201904      43960.0  37252.0   1703.0  6643.0  26441.0       2236.0  
201905      51758.0  22235.0   1129.0  4920.0   9070.0       3528.0  
201906      48418.0  23653.0    751.0  3966.0   6263.0       2784.0  
201907      47256.0   8389.0    976.0  4316.0   5889.0       2824.0  
201908      58593.0  13690.0   1512.0  6209.0  10216.0       3800.0  
201909      32470.0      0.0    674.0  2911.0   5131.0       2836.0  
201910      46641.0      0.0   1447.0  4221.0  20487.0       3336.0  
201911      45952.0      0.0    899.0  4548.0   6016.0       2368.0  
201912      41066.0      0.0      0.0  3241.0   5975.0       1732.0

In [293]:
daejun_19.columns

Index(['상소오토캠핑장', '찬샘마을', '한밭교육박물관', '대전 아쿠아월드', '대전오월드', '뿌리공원', '남선공원종합체육관',
       '대전시립미술관', '장태산자연휴양림', '계룡산 수통골', '국립중앙과학관', '대전시민천문대', '대전어린이회관',
       '솔로몬 로파크', '유성유스호스텔', '지질박물관', '화폐박물관', '로하스가족공원 캠핑장'],
      dtype='object')

In [279]:
# 2011년 1월부터 2019년 12월까지 전부 있는 칼럼
daejun_1=daejeon.iloc[:, 58:]
daejun_1=daejun_1.dropna(axis=1)
daejun_1

한밭교육박물관     대전오월드   계룡산 수통골   국립중앙과학관   지질박물관
year_month                                               
201101        100.0    7335.0   25330.0   79251.0  7375.0
201102        226.0   26175.0   34116.0   73591.0  3635.0
201103        516.0   65177.0   38547.0   57815.0  2789.0
201104        211.0  171794.0   36226.0  186133.0  6840.0
201105        360.0  213740.0   38242.0  163611.0  6690.0
...             ...       ...       ...       ...     ...
201908       1647.0   57721.0   85577.0  158881.0  6209.0
201909       5491.0   66082.0   78944.0  157633.0  2911.0
201910       2813.0  148600.0  119600.0  137386.0  4221.0
201911       2258.0   52131.0  107300.0   75532.0  4548.0
201912       2312.0   26374.0   72600.0   73972.0  3241.0

[108 rows x 5 columns]

In [285]:
tour=daejeon[['상소오토캠핑장', '찬샘마을', '한밭교육박물관', '대전 아쿠아월드', '대전오월드', '뿌리공원', '남선공원종합체육관',
       '대전시립미술관', '장태산자연휴양림', '계룡산 수통골', '국립중앙과학관', '대전시민천문대', '대전어린이회관',
       '솔로몬 로파크', '유성유스호스텔', '지질박물관', '화폐박물관', '로하스가족공원 캠핑장']]

In [289]:
daejeon=daejeon.iloc[:, :58]

In [291]:
daejeon=pd.concat([daejeon, tour], axis=1)

18개의 관광지를 대상으로 분석해보기
--

우선, 2011년 1월부터 2019년 12월까지 데이터가 있는 한밭교육박물관, 대전오월드, 계룡산 수통골, 국립중앙과학관, 지질박물관부터 분석해보도록 하겠습니다.